### PR TITLE
fix(valahighlight): fix the break block and improve the highlights

### PR DESCRIPTION
**Pull request by Decodetalkers**
image
Finally

### CODE REVIEW

**Comment by Decodetalkers** [comment]
If meet highlight problem , it is the problem of treesitter.. not mime.. :) the treesitter still have a lot of problem, I just fix and improve the highlight...

**Comment by theHamsta** [comment]
> meet highlight problem , it is the problem of treesitter.. not mime.. :) the treesitter still have a lot of problem, I just fix and improve the highlight...

I don't understand. Did you discover anything specific?

**Comment by theHamsta** [comment]
@matbme

**Comment by Decodetalkers** [comment]
> 

I just test a snippet, and with playground I found a lot of error, so the highlight works very bad.. but I have test the repo upstream , it works well

File: queries/vala/folds.scm [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[
+  (namespace)
+	(function_definition)
+	(class_declaration)
+	(interface_declaration)
+] @fold
+
```
Comment (theHamsta) [inline queries/vala/folds.scm:7]: Nice, that this got added!

**Comment by Decodetalkers** [comment]
image
As you see, There is a lot of error

File: queries/vala/highlights.scm [hunk 2]
```diff
@@ -54,46 +58,56 @@
...
 [
   "typeof"
   "is"
 ] @keyword.operator
 
+(this) @variable.builtin
+;;; move enum to namespace
+[
+  "enum"
+] @namespace
```
Comment (theHamsta) [inline queries/vala/highlights.scm:72]: I don't think this is `@namespace`

**Comment by Decodetalkers** [comment]
```vala


namespace Lsp {
    [CCode (default_value = "LSP_TEXT_DOCUMENT_SYNC_KIND_Unset")]
    enum TextDocumentSyncKind {
        Unset = -1,

        None = 0,

        Full = 1,

        Incremental = 2
    }
    class DocumentSymbol : Object, Json.Serializable {
        private Vala.SourceReference? _source_reference;
        public string name { get; set; }
        public string? detail { get; set; }
        public SymbolKind kind { get; set; }
        public bool deprecated { get; set; }
        private Range? _initial_range;
        public Range range {
            owned get {
                if (_initial_range == null)
                    _initial_range = new Range.from_sourceref (children.first ()._source_reference);
                
                return children.fold<Range> ((child, current_range) => current_range.union (child.range), _initial_range);
            }
        }
        public Range selectionRange { get; set; }
        public Gee.List<DocumentSymbol> children { get; private set; default = new Gee.LinkedList<DocumentSymbol> (); }
        public string? parent_name;

        private DocumentSymbol () {}

        /**
         * @param type the data type containing this symbol, if there was one (not available for Namespaces, for example)
         * @param sym the symbol
         */

        public new void Json.Serializable.set_property (ParamSpec pspec, Value value) {
            base.set_property (pspec.get_name (), value);
        }

        public new Value Json.Serializable.get_property (ParamSpec pspec) {
            Value val = Value (pspec.value_type);
            base.get_property (pspec.get_name (), ref val);
            return val;
        }

        public unowned ParamSpec? find_property (string name) {
            return this.get_class ().find_property (name);
        }

        public Json.Node serialize_property (string property_name, Value value, ParamSpec pspec) {
            if (property_name != "children")
                return default_serialize_property (property_name, value, pspec);
            var node = new Json.Node (Json.NodeType.ARRAY);
            node.init_array (new Json.Array ());
            var array = node.get_array ();
            foreach (var child in children)
                array.add_element (Json.gobject_serialize (child));
            return node;
        }

        public bool deserialize_property (string property_name, out Value value, ParamSpec pspec, Json.Node property_node) {
            error ("deserialization not supported");
        }
    }
}
```
snippet is here, you can test yourself, but with the repo upstream there is no error,have you synchronize with https://github.com/vala-lang/tree-sitter-vala ?

File: queries/vala/highlights.scm [hunk 3]
```diff
@@ -54,46 +58,56 @@
...
+;;; conditional
 [
   "if"
   "else"
   "switch"
   "case"
   "default"
+  "throws"
+	"try"
+  "catch"
```
Comment (theHamsta) [inline queries/vala/highlights.scm:110]: There is `@exception` for those three

Comment (Decodetalkers) [reply queries/vala/highlights.scm:113]: > There is `@exception` for those three

fixed

File: queries/vala/highlights.scm (hunk 2 above)
Comment (Decodetalkers) [reply queries/vala/highlights.scm:72]: > I don't think this is `@namespace`

What is your suggestion, I think it should be another color..or just put it in keyword?

Comment (Decodetalkers) [reply queries/vala/highlights.scm]: > I don't think this is `@namespace`

now it has returned to keyword.

**Comment by Decodetalkers** [comment]
I see, there is something error with 
https://github.com/matbme/tree-sitter-vala, and you use this repo..

**Comment by matbme** [comment]
Hmm, AFAIK this is all fixed in [https://github.com/vala-lang/tree-sitter-vala](). Maybe it's time to bring back #2818.

**Comment by Decodetalkers** [comment]
> Hmm, AFAIK this is all fixed in https://github.com/vala-lang/tree-sitter-vala. Maybe it's time to bring back #2818.

I wonder if the key words will be the same, either all scheme file will be rewrite..

**Comment by clason** [comment]
Yeah, that PR _will_ be merged, so it's best to park any work on Vala queries until then.

**Comment by clason** [comment]
It's merged now, so you'll have to rebase (but check the new highlight first whether they already address your problems).

I'm taking the liberty as marking this PR as draft while you do that!

**Pull request closed without merging**